experimental/packages/opentelemetry-instrumentation/src/semver.ts

```diff
@@ -0,0 +1,585 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
As being discussed in a separate thread (on the test file), we will need to decide on the license/copyright.

Is this implementation significantly copied from node-semver?
That's relevant both for license/copyright and for helping review this.  I'm not sure how carefully I should review this (it is a fair amount of code). Passing all the semver.satisfies tests is very nice.

Reply:
@trentm 
The following methods have been copied with **very minor** modifications:
- `replaceTilde`
- `replaceCaret`
- `replaceXRange`
- `replaceHyphen`
- regexp definitions used by them

Reply:
@trentm 

I have also another custom semver implementation (without any code part is imported from actual `semver` package, but some parts of it is a little bit hacky) and all `semver.satisfies` tests are also passing with it too. But the problem is that there are many specific rules by `semver` package and some of them are not explicitly mentioned in the README and hidden in their code. 

So, even though `semver.satisfies` all tests are passing, I am not very confident about how my other custom semver implementation is compatible with actual `semver` package if there are cases which are not tested.